core/drivers/crypto/caam/caam_jr.c

```diff
@@ -0,0 +1,781 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM Job Rings manager.
+ *         Implementation of functions to enqueue/dequeue CAAM Job Descriptor
+ */
+
+/* Global includes */
+#include <kernel/interrupt.h>
+#include <kernel/panic.h>
+#include <kernel/pm.h>
+#include <kernel/spinlock.h>
+#include <mm/core_memprot.h>
+#include <tee/cache.h>
+
+/* Local includes */
+#include <caam_common.h>
+#include <caam_io.h>
+#include <caam_jr.h>
+#include <caam_hal_jr.h>
+#include <caam_rng.h>
+#include <caam_utils_delay.h>
+#include <caam_utils_mem.h>
+
+/*
+ * Enable the interrupt mode or not
+ */
+#ifdef CFG_CRYPTO_DRIVER
+//#define IT_MODE
+#endif
+
+/*
+ * Definition of input ring object
+ */
+struct inring_entry {
+	descEntry_t desc; /* Physical address of the descriptor */
+};
+
+/*
+ * Definition of output ring object
+ */
+struct __attribute__((__packed__)) outring_entry {
```

Comment:
Please use the `__packed` macro from `<compiler.h>` instead.

Reply:
ok